secrets/secrets.go

```diff
@@ -293,42 +302,56 @@ func (sg *SecretGenerator) generateSecret(manifest model.Manifest, secrets *v1.S
 			return fmt.Errorf("Can't convert `%s` generator config into JSON: %s", configVar.Name, err)
 		}
 
-		// if generator input has changed, then the secret needs to be regenerated
 		name := util.ConvertNameToKey(configVar.Name)
-		if !bytes.Equal(secrets.Data[name+generatorSuffix], generatorInput) {
+		// if secret exists and generator input has changed, then the secret needs to be regenerated
```

Comment:
Why do we not need to regenerate if the secret doesn't exist?

Reply:
Oh, because we generate anyway if they don't exist.

Reply:
Yes, this check just suppresses the misleading progress message that the secret needs to be regenerated.